### DISCUSSION TITLE
Have 2 questions

### DISCUSSION THREAD

**Discussion by ad203amiar** [linux-surface/linux-surface#1372] [Q&A]
I have installed linux for surface on surface pro 1 running ubunta 22.04.
I missed the first time that grub came up with the blue screen and i did not ener the password, now it doesn't show the blue screen, how can I get it back or put the password?
Although I have installed Libwacom the screen does not turn, what am i doing wrong?

**Comment by vistis** [comment]
Regarding your second question, do you mean that your screen does not rotate? I don't think `libwacom` is related to that functionality as far as I know. I think it is just to make the tablet functionalities (ie, pen input) work. Also, I recommend installing `libwacom-surface` instead, as it will improve compatibility.

**Comment by ad203amiar** [comment]
Yes, i should have mentioned I installed Libwacom Surface and yes sreen does not rotate when Surface is rotated.

**Reply by vistis** [reply]
I see. Mind checking if auto rotate is on? You might need to disconnect your keyboard to see the option in the quick setting menu.

**Reply by ad203amiar** [reply]
Sorry, I'm very new to linux, I tran uname-a again and it is linux.Surface, nothing about rotation, with or without keyborad shows in quick settings or settins menu. Also nothing happens when i detach keyboard, do not see any online keybords popping up, though I have enabled it.

**Reply by vistis** [reply]
Hmm, it shouldn't be an issue if you have followed the guide... It might be a SP1 specific issue for all I know. It works fine on my SP8, running GNOME 45.3 Wayland on Arch Linux. The only thing that might be responsible for it might be GNOME, which is the default desktop environment of Ubuntu. I suggest using a `Wayland` session instead of and `Xorg` session. You should be able to select it when you first login by clicking the cog icon in the bottom right corner.

I have found a discussion about this issue that might worth a look: https://gitlab.gnome.org/GNOME/mutter/-/issues/1686

**Reply by ad203amiar** [reply]
I have fixed that by installing Gnome shell extentions and installing Screen rotate from Gnome shell extentions. Thanks for all the help :)

**Comment by ad203amiar** [comment]
There is only 1 questions left ,How do i fix the grub,

**Comment by ad203amiar** [comment]
From: https://www.reddit.com/r/SurfaceLinux/comments/hkntd7/secure_boot_problems/
 sudo apt remove linux-surface-secureboot-mok

sudo apt install linux-surface-secureboot-mok

reboot 

After that I had another problem Keyboard did not work in to register mok.
And from : https://unix.stackexchange.com/questions/608217/keyboard-does-not-work-in-mokmanager-during-key-enrollment
TLDR: Enter BIOS, this enabled the keyboard in MOK Manager for me. You don't need to change any setting there, you can directly exit it after entering it.


I had the same problem after installing Linux Mint 20.1 on my Lenovo Legion 5 Pro 16ACH6H.

After MOK Manager started it didn't recognize my keyboard. I had to shortly press the power button to turn it of. Now I turned it on, pressed F2 to get into my BIOS (the keyboard is working there). I directly left the BIOS without changing anything and voilà - the keyboard was working then!
